### PR TITLE
Support for private repos

### LINKED ISSUE #26: [IDB + MDB] Feature request: support for private repositories
Due to how repos are cloned by IDB and MDB, it's not possible for them to create diffs for private git respositories. I'd like to just hand an access token and have that squared away. @Cyberboss

### PR DESCRIPTION
**Pull request by Cyberboss**
- Use GitHub API for downloading files.
- Use installation token when cloning and fetching repositories.
- Remove unused function.

Untested, tell me how.

Closes #26

### CODE REVIEW

**Review by Cyberboss — COMMENTED** [review]
I need to update the remote to use the current token when fetching too otherwise the original could expire

**Comment by optimumtact** [comment]
pretty sure you can have ilke 1 free privat repo on github these days for testing purposes.

**Comment by AffectedArc07** [comment]
> pretty sure you can have ilke 1 free privat repo on github these days for testing purposes.

You can have infinite

**Comment by Cyberboss** [comment]
Created some local test cases. Should work fine without issue

File: crates/mapdiffbot2/src/runner.rs [hunk 1]
```diff
@@ -116,7 +116,7 @@ async fn job_handler(name: &str, job: Job, blob_client: Azure) {
 
     let output = actix_web::rt::time::timeout(
         Duration::from_secs(7200),
-        actix_web::rt::task::spawn_blocking(move || do_job(job, blob_client)),
```
Comment (jupyterkat) [inline crates/mapdiffbot2/src/runner.rs:119]: Keep the spawn blocking

File: crates/mapdiffbot2/src/job_processor.rs [hunk 2]
```diff
@@ -331,7 +333,7 @@ fn generate_finished_output<P: AsRef<Path>>(
     Ok(builder.build())
 }
 
-pub fn do_job(job: Job, blob_client: Azure) -> Result<CheckOutputs> {
+pub async fn do_job(job: Job, blob_client: Azure) -> Result<CheckOutputs> {
```
Comment (jupyterkat) [inline crates/mapdiffbot2/src/job_processor.rs:336]: Don't do async here

File: crates/mapdiffbot2/src/job_processor.rs [hunk 3]
```diff
@@ -342,6 +344,11 @@ pub fn do_job(job: Job, blob_client: Azure) -> Result<CheckOutputs> {
 
     let base = &job.base;
     let head = &job.head;
+
+    let (_, secret_token) = octocrab::instance()
+        .installation_and_token(job.installation)
+        .await?;
```
Comment (jupyterkat) [inline crates/mapdiffbot2/src/job_processor.rs:350]: Use `block_on`

**Review by jupyterkat — CHANGES_REQUESTED** [review]

**Comment by Cyberboss** [comment]
This fucked up in prod so I gotta figure that out too

**Comment by Cyberboss** [comment]
Tested and working!

**Comment by DisturbHerb** [comment]
worth the weight 🏋️‍♂️

**Review by jupyterkat — APPROVED** [review]

File: crates/diffbot_lib/src/github/github_api.rs [hunk 4]
```diff
@@ -244,34 +240,57 @@ pub async fn download_url<S: AsRef<str>>(
...
-    path.push(&target.sha);
-    path.set_extension("dmi"); // Method should have an IDB qualifier due to being a shared crate
-
-    async_fs::create_dir_all(path.parent().unwrap()).await?;
-    let mut file = File::create(&path).await?;
-
-    let data = download_url(installation, repo, &filename, &commit).await?;
-    file.write_all(&data).await?;
-    Ok(path)
+/* local test requires commenting out the .installation(...) call in find_content(), a valid github token with access, and the following dep: actix-rt = "2.9.0"
```
Comment (AffectedArc07) [inline crates/diffbot_lib/src/github/github_api.rs:252]: Should we remove this?

Comment (jupyterkat) [reply crates/diffbot_lib/src/github/github_api.rs:252]: nah, it could be used for reference for testing

**Pull request merged**